src/main/java/org/myrobotlab/service/AzureTranslator.java

```diff
@@ -12,96 +12,118 @@
 import org.myrobotlab.framework.Service;
 import org.myrobotlab.logging.Level;
 import org.myrobotlab.logging.LoggerFactory;
-import org.myrobotlab.logging.Logging;
 import org.myrobotlab.logging.LoggingFactory;
 import org.myrobotlab.service.interfaces.TextListener;
 import org.myrobotlab.service.interfaces.TextPublisher;
 import org.slf4j.Logger;
 
-import io.github.firemaples.detect.Detect;
-import io.github.firemaples.language.Language;
-import io.github.firemaples.translate.Translate;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
 
 public class AzureTranslator extends Service implements TextListener, TextPublisher {
 
   private static final long serialVersionUID = 1L;
 
-  String toLanguage = "it";
-  String fromLanguage = null;
-  public final static Logger log = LoggerFactory.getLogger(AzureTranslator.class);
+  /**
+   * key field name for security service
+   */
+  public final static String AZURE_TRANSLATOR_KEY = "azure.translator.key";
 
-  public static void main(String[] args) throws Exception {
-    LoggingFactory.init(Level.INFO);
-    try {
+  /**
+   * language to translate to
+   */
+  String to = "fr";
 
-      AzureTranslator translator = (AzureTranslator) Runtime.start("translator", "AzureTranslator");
-      Runtime.start("gui", "SwingGui");
-      log.info("Translator service instance: {}", translator);
+  /**
+   * language to translate from
+   */
+  String from = "en";
 
-    } catch (Exception e) {
-      Logging.logError(e);
-    }
-  }
+  /**
+   * location
+   */
+  String location = "eastus";
 
-  public AzureTranslator(String n, String id) {
-    super(n, id);
+  transient OkHttpClient client = new OkHttpClient();

```

Comment:
Why can't we just use apache httpclient?  Seems silly to include yet another http client library for a straight forward http request.

Reply:
because okhttp is the better library  (also more cloud services are providing examples with okhttp)

https://stackoverflow.com/questions/42392778/okhttp-or-httpclient-which-offers-better-functionality-and-more-efficiency